src/vm.rs

```diff
@@ -2359,14 +2397,13 @@ impl<'a, T: GuestStagePagingMode> FinalizedVm<'a, T> {
                     self.unshare_mem_region(addr, len)
                 };
 
-                // Block if we need a TLB invalidation.
+                // Always block given we expect a TLB increment to be triggered from the host.
                 let action = match result {
-                    Ok(tlbv) if tlbv > self.vm_pages().min_tlb_version() => EcallAction::Break(
-                        VmExitCause::BlockingEcall(SbiMessage::TeeGuest(guest_func), tlbv),
-                        SbiReturn::success(0),
-                    ),
                     Ok(_) => EcallAction::Break(
-                        VmExitCause::ResumableEcall(SbiMessage::TeeGuest(guest_func)),
+                        VmExitCause::BlockingEcall(
+                            SbiMessage::TeeGuest(guest_func),
+                            self.vm_pages().min_tlb_version(),
+                        ),
```

Comment:
now that the host is responsible for explicitly blocking & removing the pages, we should remove this logic to block until a specific tlb version is met.

Reply:
i think we can drop the `Blocked` state entirely now that we handle that logic in `complete_pending_op()`.

Reply:
Sounds good